src/tigerbeetle/cli.zig

```diff
@@ -915,7 +922,7 @@ fn parse_args_start(start: CLIArgs.Start) Command.Start {
     const lsm_forest_node_count: u32 =
         @intCast(@divExact(lsm_manifest_memory, constants.lsm_manifest_node_size));
 
-    return .{
+    var command_start: Command.Start = .{
```

Comment:
Open to better ways of managing the buffer for generated aof path for `--aof`!

Reply:
(I figured we'd want to keep all backwards compatibility CLI logic inside cli.zig...?)

Reply:
I think this won't work, the Start structure can move, invalidating interior pointer. You'll have to use a helper function here:

```zig
pub const Command = union(enum) {
    pub const Start = struct {
        aof_file: ?[:0]const u8,
        aof_file_buffer: [std.fs.max_path_bytes]u8 = .{0} ** std.fs.max_path_bytes,
    
        pub fn get_aof_path(start: *const Start) ?[:0]const u8 {
            if (start.aof_file) |file| return file;
            if (start.aof_file_buffer[0] != 0) {
                const index = std.mem.indexOfScalar(u8, start.aof_file_buffer, 0).?;
                assert(index > 0);
                return &start.aof_file_buffer[0..index :0];
            }
        }
    };

```

Reply:
Almost everywhere yes, and done in [bae3791](https://github.com/tigerbeetle/tigerbeetle/pull/3055/commits/bae3791b120084f4bd9480a701ed62eb1c8b2de6)!